### ISSUE TITLE
gli-version

### ISSUE THREAD

**Issue by KwadroNaut** [jedcn/reveal-ck#77]
I had a newer gli-gem installed (2.14.0), but 2.13.2 is expected from reveal-ck. If I uninstall 2.13.2, or have both installed, reveal-ck breaks:

/usr/lib/ruby/2.3.0/rubygems/specification.rb:2287:in `raise_if_conflicts': Unable to activate reveal-ck-3.4.0, because gli-2.14.0 conflicts with gli (= 2.13.2) (Gem::ConflictError)

Is this expected, or is there some reason to require the specific older 2.13.2?

**Comment by jedcn** [comment]
Hey there!

Thanks for taking the time to speak up!

I hope you're liking reveal-ck and it's useful to you.

It's expected (by me) that the versions are locked tightly in that `reveal-ck` is more like an 'app' than it is a 'library.'

There was a time when I let the versions vary until a major version change, but I encountered bugs from combinations of libraries interacting with each other.

My solution was to lock the versions quite tightly and suggest that people use a `Gemfile` whenever using `reveal-ck`.

The `Gemfile` would include `reveal-ck` and once you ran a `bundle` the `Gemfile.lock` would hold the required version of `gli` and so when you ran commands with `bundle exec reveal-ck` you'd get the right version even if you had 10 versions installed.

When I look at my setup, I have two versions of `gli` installed like you do, but things are working for me:

```
$ gem list | grep gli
gli (2.14.0, 2.13.2)
```

If I uninstall gli 2.13.2 then it fails (as you'd expect):

```
reveal-ck --version
/opt/rubies/2.2.4/lib/ruby/2.2.0/rubygems/dependency.rb:315:in `to_specs': Could not find 'gli' (= 2.13.2) - did find: [gli-2.14.0] (Gem::LoadError)
```

And when I installed it again things are all good:

```
$ gem install gli -v 2.13.2
Fetching: gli-2.13.2.gem (100%)
Successfully installed gli-2.13.2
1 gem installed
$ reveal-ck --version 
reveal-ck version 3.4.0
```

---

`gli` version `2.13.2` dates back to more than a year and a half ago.

I periodically update all of the libraries to the latest version-- let me take a look at doing that.

If we go that route, I'd open a PR with new library versions and publish a new version of reveal-ck. The new version of `reveal-ck` would use the latest version of `gli`.

I'll update this issue if/when that happens.
